conans/test/functional/toolchains/meson/test_meson.py

```diff
@@ -167,3 +170,57 @@ def test(self):
         # res/tutorial -> tutorial is being added automatically by Meson
         assert os.path.exists(os.path.join(package_folder, "res", "tutorial", "file1.txt"))
         assert os.path.exists(os.path.join(package_folder, "res", "tutorial", "file2.txt"))
+
+
+@pytest.mark.tool_meson
+def test_meson_and_user_filenames_composition():
+    """
+    Testing when users wants to append their own meson files and override/complement some
+    sections from Conan file ones.
+
+    See more information in Meson web page: https://mesonbuild.com/Machine-files.html
+
+    In this test, we're overriding only the Meson section ``[binaries]`` for instance.
+    """
+    profile = textwrap.dedent("""
+        [settings]
+        os=Windows
+        arch=x86_64
+        compiler=gcc
+        compiler.version=9
+        compiler.cppstd=17
+        compiler.libcxx=libstdc++11
+        build_type=Release
+
+        [conf]
+        tools.meson.mesontoolchain:user_filenames=["myfilename.ini"]
+   """)
+    myfilename = textwrap.dedent("""
+    [binaries]
+    c = '/usr/fake/path/gcc'
```

Comment:
The definition of compilers might be "standardized" by Conan in its own configuration (or handle env-var CC CXX for everyone), so maybe not a good example